internal/event/target/kafka.go

```diff
@@ -268,14 +269,30 @@ func (target *KafkaTarget) Close() error {
 }
 
 // Check if atleast one broker in cluster is active
-func (k KafkaArgs) pingBrokers() bool {
-	d := net.Dialer{Timeout: 60 * time.Second}
-	for _, broker := range k.Brokers {
-		if _, err := d.Dial("tcp", broker.String()); err == nil {
-			return true
+func (k KafkaArgs) pingBrokers() (err error) {
+	d := net.Dialer{Timeout: 1 * time.Second}
+
+	errs := make([]error, len(k.Brokers))
+	var wg sync.WaitGroup
+	for idx, broker := range k.Brokers {
+		broker := broker
+		idx := idx
+		wg.Add(1)
+		go func(broker xnet.Host, idx int) {
+			defer wg.Done()
+
+			_, errs[idx] = d.Dial("tcp", broker.String())
+		}(broker, idx)
+	}
+	wg.Wait()
+
+	var retErr error
+	for _, err := range errs {
+		if err != nil {
+			retErr = err
```

Comment:
This doesn't match the function signature. It says it will succeed if *one* broker is reachable. This means that *all* brokers must be reachable.

Without too much info, I think one should be fine. But YMMV.